homeassistant/components/media_player/enigma2.py

```diff
@@ -0,0 +1,220 @@
+"""
+Support for Enigma2 based media players.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.enigma2/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, SUPPORT_TURN_ON,
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    MediaPlayerDevice, PLATFORM_SCHEMA, MEDIA_TYPE_TVSHOW)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON, STATE_UNKNOWN, CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['openwebif.py==0.9']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TIMEOUT = 'timeout'
+
+DEFAULT_NAME = 'Enigma2 Media Player'
+DEFAULT_PORT = 80
+DEFAULT_TIMEOUT = 0
+
+# Support different items when watching recording playback
+SUPPORT_ENIGMA2_RECORDING_PLAYBACK = SUPPORT_VOLUME_SET | \
+                                     SUPPORT_VOLUME_MUTE | \
+                                     SUPPORT_TURN_OFF | SUPPORT_PAUSE
+
+SUPPORT_ENIGMA2_LIVE_TV = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+                          SUPPORT_TURN_OFF | SUPPORT_NEXT_TRACK | \
+                          SUPPORT_PREVIOUS_TRACK | SUPPORT_TURN_ON
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Enigma2 TV platform."""
+    name = config.get(CONF_NAME)
```

Comment:
Don't use `dict.get(key)` on required config keys or keys that have default values. Use `dict[key]`.